tokio/src/io/async_read_ext.rs

```diff
@@ -60,6 +61,14 @@ pub trait AsyncReadExt: AsyncRead {
     {
         read_exact(self, dst)
     }
+
+    /// Read all bytes until EOF in this source, placing them into `dst`.
+    fn read_to_end<'a>(&'a mut self, dst: &'a mut Vec<u8>) -> ReadToEnd<'a, Self>
```

Comment:
I'm curious if you can elide this lifetime?

Reply:
no, because the arg also requires the lifetime. I think eliding only works when it is only for self + ret type.